Add form submission backend [WIP]

dinoocch opened the pull request:
This commit adds the ability to submit a form via the web interace and
parse the resulting json data into a model.  In addition, we rely on the
model validation to check the forms correctness.

- [X]  Form Submission
- [ ]  Redirection
- [ ]  Error Handling

dinoocch committed:
Add form submission backend

This commit adds the ability to submit a form via the web interace and
parse the resulting json data into a model.  In addition, we rely on the
model validation to check the forms correctness.

dinoocch commented:
@m-wynn Are we using the same snack bar style for error messages as in Orderboard?

m-wynn commented:
@dinoocch sure, that's fine, unless you have a better idea.

dinoocch commented:
On form submit, the view will redirect the user to either:
- [ ] a list of forms
- [ ] the form submission page for the created form
- [x] the rendered form preview
* Not sure which is better...

the-maldridge commented:
I'm inclined to redirect to a page that shows the rendered version of what was just entered.  Probably the same page we'd use for printing since then we can kill two birds with one stone.

m-wynn commented on constellation_forms/views.py:
```diff
@@ -1,16 +1,70 @@
 
+    def post(self, request):
+        ''' Creates a form '''
+        form_data = json.dumps(request.body)
```
Shouldn't this be `form_data = json.loads(request.POST['data'])`?

dinoocch committed:
Add initial list views

This commit adds an initial list of forms and form submissions.
Permissions still pending.

m-wynn commented:
json.dumps still needs to be json.loads

dinoocch commented:
Woops, you're correct, I will fix when I get back... >.<
